### PR TITLE
Attempt to disallow optimade versions<0.10

### LINKED ISSUE #35: Validator action should fail nicely for validator versions below 0.10
Since v0.10 of optimade-python-tools, the way of calling the validator has changed from `optimade_validator` to `optimade-validator`, so requesting a version <0.10 causes failures. It's not worth us supporting the older version, so we should simply raise an error if the version requested is too low.

### LINKED ISSUE #38: Cannot install double digit version
The regular expressions for determining whether the `validate_version` is an actual version number, e.g., `0.10.0`, or `v0.10.1` do not allow for double digits, i.e., `0.10.0` is not recognized, while `0.9.0` is.

### PR DESCRIPTION
**Pull request by ml-evs**
Closes #35 
Fixes #38

### CODE REVIEW

**Comment by CasperWA** [comment]
Ah, yeah, for the version tests you should probably just update them to expect a failure :)

**Comment by CasperWA** [comment]
I found the issue here, see my opened issue #38 - I assigned it to you to fix in this PR, it's basically just adding `+` after the int ranges.

File: entrypoint.sh [hunk 1]
```diff
@@ -87,6 +87,13 @@ esac
 
 
 API_VERSION=($(python -c "from optimade import __api_version__; versions = [__api_version__.split('-')[0].split('+')[0].split('.')[0], '.'.join(__api_version__.split('-')[0].split('+')[0].split('.')[:2]), '.'.join(__api_version__.split('-')[0].split('+')[0].split('.')[:3])]; print(' '.join(versions))"))
+PACKAGE_VERSION=($(python -c "from optimade import __version__; print(__version__.replace('.', ' '))"))
+
+if [ ${PACKAGE_VERSION[0]} -eq 0 ] && [ ${PACKAGE_VERSION[1]} -lt 10 ]; then
+    echo "Incompatible validator version requested ${INPUT_VALIDATOR_VERSION}, please use >=0.10."
+    exit 1
+fi
+
```
Comment (CasperWA) [inline entrypoint.sh:96]: To at least fail a bit faster, this should be moved up to right after having installed the `optimade` package.

Also, it's fine to keep in the original checks, although this is cleaner, of course.

**Comment by ml-evs** [comment]
Seems to be failing correctly, minus the printing of the word "ERROR" which I thought might be handled by BATS itself?

**Comment by CasperWA** [comment]
> Seems to be failing correctly, minus the printing of the word "ERROR" which I thought might be handled by BATS itself?

No, this is simply checking the printed output. I think normally if a command fails "ERROR: ..." will be printed, hence the check. But here, you're not printing `ERROR`. You could add that if you wish?

**Comment by CasperWA** [comment]
> > I think normally if a command fails "ERROR: ..." will be printed, hence the check. But here, you're not printing `ERROR`. You could add that if you wish?
> 
> What do you mean "normally"? I don't see it printed for e.g. installing v0.0.0? I think it's also falling over because we can't trigger the test for API version 0.10.1 anymore, as we don't support any optimade versions that use that...

Indeed. But this should just be updated / the test with API version v0.10.1 should just be removed, I think? There is no way now to test that this works, unfortunately.

**Comment by CasperWA** [comment]
> > I think normally if a command fails "ERROR: ..." will be printed, hence the check. But here, you're not printing `ERROR`. You could add that if you wish?
> 
> What do you mean "normally"? I don't see it printed for e.g. installing v0.0.0? I think it's also falling over because we can't trigger the test for API version 0.10.1 anymore, as we don't support any optimade-python-tools versions with `__api_version__ < 1` that use that...

Ah, the "ERROR" it output from `pip`.

**Comment by CasperWA** [comment]
> > > I think normally if a command fails "ERROR: ..." will be printed, hence the check. But here, you're not printing `ERROR`. You could add that if you wish?
> > 
> > 
> > What do you mean "normally"? I don't see it printed for e.g. installing v0.0.0? I think it's also falling over because we can't trigger the test for API version 0.10.1 anymore, as we don't support any optimade versions that use that...
> 
> Indeed. But this should just be updated / the test with API version v0.10.1 should just be removed, I think? There is no way now to test that this works, unfortunately.
> Although it's good enough to make sure that the v1.0.0-rc1 are indeed parsed as v1.0.0 for the URL :)

Also v1.0.0-rc1 is prior to v0.10.0 - the update to v1.0.0 happened in `optimade` v0.9.7.

**Review by CasperWA — APPROVED** [review]
Looks good to me :)

**Comment by CasperWA** [comment]
If you are fine with this, feel free to squash+merge, and then perhaps also release a new v2.1.0? 😃

**Pull request merged**